sphinx_needs/lsp/esbonio.py

```diff
@@ -0,0 +1,385 @@
+"""Suport Sphinx-Needs language features."""
+import getpass
+import os
+import pathlib
+import re
+from hashlib import blake2b
+from typing import List, Optional, Tuple, Union
+
+from esbonio.lsp import LanguageFeature
+from esbonio.lsp.rst import CompletionContext, DefinitionContext, HoverContext
+from esbonio.lsp.sphinx import SphinxLanguageServer
+from pygls.lsp.types import (
+    CompletionItem,
+    CompletionItemKind,
+    InsertTextFormat,
+    Location,
+    Position,
+    Range,
+    TextEdit,
+)
+
+from sphinx_needs.lsp.needs_store import NeedsStore
+
+
+class NeedlsFeatures(LanguageFeature):  # type: ignore[misc]
```

Comment:
this error occurs because mypy can't 'see' the `esbonio` types, since the esbonio package is not installed in the pre-commit environment.

There are two solutions

1. add the esbonio package to the mypy additional dependencies in `pre-commit-config.yaml`
2. modify the mypy pre-commit hook to be a 'local' hook, which is run from the dev environment, not a dedicated pre-commit environment.

I'll make a PR for option 2, since i think that's a more robust general solution.

Reply:
i've created #588 with this change.